test/cbmc/source/stubs.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * AWS IoT Over-the-air Update v3.1.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file stubs.c
+ * @brief Includes the definition of all the stubs required for the CBMC proofs.
+ */
+/*  Ota Agent includes. */
+#include "ota.h"
+#include "stubs.h"
+#include <stdlib.h>
+
+OtaErr_t initFileTransferStub( OtaAgentContext_t * agent )
+{
+    OtaErr_t err;
+
+    __CPROVER_assert( agent != NULL,
+                      "Error: OtaAgent needs to be initialized before calling initFileTransfer." );
+
+    __CPROVER_assume( ( err >= OtaErrNone ) && ( err <= OtaErrActivateFailed ) );
+
+    return err;
+}
+
+OtaOsStatus_t startTimerStub( OtaTimerId_t otaTimerId,
+                              const char * const pTimerName,
+                              const uint32_t timeout,
+                              OtaTimerCallback_t callback )
+{
+    OtaOsStatus_t status;
+
+    /* status must have values only from the OtaOsStatus_t enum. */
+    __CPROVER_assume( ( status >= OtaOsSuccess ) && ( status <= OtaOsTimerDeleteFailed ) );
+
+    __CPROVER_assert( ( otaTimerId != OtaSelfTestTimer ) || ( otaTimerId != OtaRequestTimer ),
```

Comment:
Should this not be:
```suggestion
    __CPROVER_assert( ( otaTimerId == OtaSelfTestTimer ) || ( otaTimerId == OtaRequestTimer ),
```
or am I missing something?